### PR TITLE
Listen task event in Moonbase. fix #56

### LINKED ISSUE #56: There’s no validation of the task that should trigger
There’s no validation of the event that should happen. 

Please use the `listenEvents()` function in src/common/utils.js to make sure there’s an ethereum execution on Moonbase local.

_Originally posted by @chrisli30 in https://github.com/OAK-Foundation/xcm-demo/pull/43#discussion_r1174656841_

### PR DESCRIPTION
**Pull request by imstar15**
Log:

```
5. Keep Listening XCM events on Moonbase Local until 2023-05-09 23:00:00(1683644400000) to verify that the task(taskId: 0x591c31b20aeb42e85a76875208eedce856704a9ee9595d4d1545724c99531efd, providerId: xcmp_automation_test_yelrs) will be successfully executed ...
        ethereum:Executed:: (phase={"applyExtrinsic":1})
                        H160: 0x8097c3c354652cb1eeed3e5b65fba2576470678a
                        H160: 0x970951a12f975e6762482aca81e57d5a2a4e73f4
                        H256: 0x0e943ce05f83d18bed206ff9c2af1a76d6869916e5441ffc5c72b2d81e4ee71f
                        {"_enum":{"Succeed":"EvmCoreErrorExitSucceed","Error":"EvmCoreErrorExitError","Revert":"EvmCoreErrorExitRevert","Fatal":"EvmCoreErrorExitFatal"}}: {"succeed":"Stopped"}
Task has been executed!
```

### CODE REVIEW

File: src/moonbeam/moonbase-local.js [hunk 1]
```diff
@@ -283,7 +289,19 @@ const main = async () => {
         proxyAccountId,
     });
 
-    // TODO: need to add event verification of the ethereum execution
+    // Listen XCM events on Mangata side
+    const additionalWaitingTime = 5 * 60 * 1000;
+    const currentTimestamp = moment().valueOf();
+    const executionTime = (currentTimestamp - (currentTimestamp % millisecondsInHour)) + millisecondsInHour;
+    console.log(`\n5. Keep Listening XCM events on ${parachainName} until ${moment(executionTime).format('YYYY-MM-DD HH:mm:ss')}(${executionTime}) to verify that the task(taskId: ${taskId}, providerId: ${providedId}) will be successfully executed ...`);
+    const timeout = executionTime - currentTimestamp + additionalWaitingTime;
```
Comment (chrisli30) [inline src/moonbeam/moonbase-local.js:297]: It’s hard for a reader to understand the calculation here. moment library has an existing function to calculate the start of an hour, and looks more intuitive.

For example, next hour
`const timestampNextHour = (moment().add(1, 'hour').startOf('hour')).valueOf() / 1000;`

In two hours
`const timestampNextTwoHours = (moment().add(2, 'hour').startOf('hour')).valueOf() / 1000;`

Then, you can use the timestamp to construct the schedule parameter.
`const schedule = { Recurring: { frequency, nextExecutionTime: timestampNextHour } };`

**Review by chrisli30 — CHANGES_REQUESTED** [review]

**Comment by chrisli30** [comment]
> Log:
> 
> ```
> 5. Keep Listening XCM events on Moonbase Local until 2023-05-09 23:00:00(1683644400000) to verify that the task(taskId: 0x591c31b20aeb42e85a76875208eedce856704a9ee9595d4d1545724c99531efd, providerId: xcmp_automation_test_yelrs) will be successfully executed ...
>         ethereum:Executed:: (phase={"applyExtrinsic":1})
>                         H160: 0x8097c3c354652cb1eeed3e5b65fba2576470678a
>                         H160: 0x970951a12f975e6762482aca81e57d5a2a4e73f4
>                         H256: 0x0e943ce05f83d18bed206ff9c2af1a76d6869916e5441ffc5c72b2d81e4ee71f
>                         {"_enum":{"Succeed":"EvmCoreErrorExitSucceed","Error":"EvmCoreErrorExitError","Revert":"EvmCoreErrorExitRevert","Fatal":"EvmCoreErrorExitFatal"}}: {"succeed":"Stopped"}
> Task has been executed!
> ```

Yes, this listening looks good

File: src/moonbeam/moonbase-local.js (hunk 1 above)
Comment (imstar15) [reply src/moonbeam/moonbase-local.js]: Fixed

**Review by chrisli30 — APPROVED** [review]
Okay, the restructuring is nice. The code looks cleaner now 👍

**Pull request merged**